src/util/bip158.rs

```diff
@@ -526,61 +528,48 @@ mod test {
-                vec!["ce43f8403c00606c24004bcaff1291091a7395085184312cf15f2393e131e770", "010000000155ef5e61d796ef5e772762849c1e2783038e3a51119a3f6893195924b454355b010000006a4730440220522cbbd4a9a89325854b60959780bb7424a5b0e83915ca070b4db3bf73c1566702201fd8f2c2ed4cfbc880daefdb0028532149236cb8cad15d33880b6a084822c0a10121038c617df5ad4bb4386b0a3ef2a078efde3a4aa7c28cea0f40cf930ebfccde2ca7ffffffff0290f4f700000000001976a914913bcc2be49cb534c20474c4dee1e9c4c317e7eb88acfb4f9226010000001976a9142d1b5d0c4e95afd2f33d9d73c0a614a64c19019688ac00000000"],
-                vec!["6f9ff0ce05a22b70346c0357f285c98f401a3ffe18fc858e9c37503e1ba7b766", "01000000011bd7cfb432df16c0579efee7603495e64070193d76ed1376e4229f019ddb4333010000006a473044022058dd39019df64e0e54e13b19edf1539cc1eb064f532326c0ac8e90afac02a79f022015a6355e2e43a201053e017e903fb877018bd675ee694b8f3e48a357a54987d00121037cc791eacc664941b9e70346a2812a410c989e0f36c96d8919c800832d884d23ffffffff0200497f0f000000001976a914913bcc2be49cb534c20474c4dee1e9c4c317e7eb88acd221181a010000001976a914721386b921c73e94d7af7f5db1695f91d180471088ac00000000"],
-                vec!["c52ca2fa069190af53b20a905de80debd58db8942419e7f54fba0639467809d2", "02000000000101ed30ca30ee83f13579da294e15c9d339b35d33c5e76d2fda68990107d30ff00700000000006db7b08002360b0000000000001600148154619cb0e7513fcdb1eb90cc9f86f3793b9d8ec382ff000000000022002027a5000c7917f785d8fc6e5a55adfca8717ecb973ebb7743849ff956d896a7ed04004730440220503890e657773607fb05c9ef4c4e73b0ab847497ee67b3b8cefb3688a73333180220066db0ca943a5932f309ac9d4f191300711a5fc206d7c3babd85f025eac30bca01473044022055f05c3072dfd389104af1f5ccd56fb5433efc602694f1f384aab703c77ac78002203c1133981d66dc48183e72a19cc0974b93002d35ad7d6ee4278d46b4e96f871a0147522102989711912d88acf5a4a18081104f99c2f8680a7de23f829f28db31fdb45b7a7a2102f0406fa1b49a9bb10c191fd83e2359867ecdace5ea990ce63d11478ed5877f1852ae81534220"]
-            ]
-        ];
-        let ref blocks = data[0];
-        let ref txs = data[1];
-        for t in 0..7 {
-            let mut txmap = HashMap::new();
-            let ref test_case = blocks[t];
-            let block_hash = sha256d::Hash::from_hex(test_case[1]).unwrap();
-            let previous_header_hash = sha256d::Hash::from_hex(test_case[3]).unwrap();
-            let header_hash = sha256d::Hash::from_hex(test_case[5]).unwrap();
-            let block: Block = deserialize(hex::decode(test_case[2]).unwrap().as_slice()).unwrap();
+
+        // test vectors from: https://github.com/jimpo/bitcoin/blob/c7efb652f3543b001b4dd22186a354605b14f47e/src/test/data/blockfilters.json
+        //["Block Height,Block Hash,Block,[Prev Output Scripts for Block],Previous Basic Header,Basic Filter,Basic Header,Notes"]
+        let data = "[\
```

Comment:
You can make the string with `r#" json here "#` and you won't have to escape all the double quotes.
https://rahul-thakoor.github.io/rust-raw-string-literals/

Reply:
modified